### PR TITLE
Architectural toolkit

### PR DESCRIPTION
**Pull request by frederic-schwarz**
- support for architectural elements within a-frame
- scene structure to A-Frame conversion ( lots of fixes )
- A-Frame elements to scene structure conversion
- scene structure validation playground
- minimap component
- SVG floor plan export

### CODE REVIEW

**Comment by tomas-polach** [comment]
massive!! :) will review tomorrow

**Comment by abiro** [comment]
I've started using the 3dio build from the architectural-toolkit branch, because it fixes some floor display problems. 🤘 Let me know please before you delete branch.

File: src/scene/export-svg.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import callService  from '../utils/services/call.js'
+
+export default function exportSvg (args) {
+  if (!args.sceneStructure || typeof args.sceneStructure !== 'object') {
+    return Promise.reject('Svg export failed: invalid input')
+  }
+  return callService('Scene.exportSvg', {arguments: args})
+    .then(function(result) {
```
Comment (tomas-polach) [inline src/scene/export-svg.js]: this is redundant. lines 8 - 10 can be deleted

File: src/scene/structure/validate/by-type/box.js [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+      min: 0.01,
+      description: 'length'
     },
-    h: { // height in meters
+    w: { // width in meters
       type: 'number',
       defaultValue: 1,
       optional: false,
-      min: 0.01 // 1cm
+      min: 0.01, // 1cm
```
Comment (tomas-polach) [inline src/scene/structure/validate/by-type/box.js]: suggest to replacing here "// 1cm" with "// m"

**Comment by tomas-polach** [comment]
suggestions:
- [ ] rename to "architecture toolkit"
- [ ] add "v" property for versioning to every element
- [ ] make complex array params consistent in aframe (i.e. https://github.com/archilogic-com/3dio-js/blob/0191bf01071d07477d40b3fc631dc118038b7e23/src/scene/structure/validate/by-type/polyfloor.js#L16)
- [ ] `gulp jshint` trows warnings about undefined ÌO3D`variable in multiple files (i.e. closet.js)

**Pull request merged**